contracts/modules/LoanSettings/LoanSettings.sol

```diff
@@ -93,6 +95,99 @@ contract LoanSettings is State, InterestHandler, LoanSettingsEvents {
         }
     }
 
+    function migrateLoanParamsList(
+        address owner,
+        uint256 start,
+        uint256 count)
+        external
+        onlyGuardian
+    {
+        EnumerableBytes32Set.Bytes32Set storage set = userLoanParamSets[owner];
+        uint256 end = start.add(count).min256(set.length());
+        if (start >= end) {
+            return;
+        }
+        count = end-start;
+
+        bytes32 loanParamId;
+        LoanParams memory loanParamsLocal;
+
+        for (uint256 i = start; i < end; ++i) {
+            loanParamsLocal = loanParams[set.get(i)];
+            loanParamId = getLoanParam(
+                loanParamsLocal.loanToken,
+                loanParamsLocal.collateralToken,
+                loanParamsLocal.maxLoanTerm == 0 // isTorqueLoan
+                    ? true
+                    : false
+            );
+            loanParamsIds[loanParamId] = loanParamsLocal.id;
+        }
+    }
+
+    function getLoanParam(
+        address loanToken,
+        address collateralToken,
+        bool isTorqueLoan)
+        pure
+        internal
+        returns(bytes32)
+    {
+        return keccak256(abi.encodePacked(
+                    loanToken,
+                    collateralToken,
+                    isTorqueLoan
+                ));
+    }
+
+    // This function intends to be PUBLIC so that anyone can create params if loanToken and collateralToken are approved by the protocol
+    function createDefaultParams(
+        address loanToken,
+        address collateralToken,
+        bool isTorqueLoan)
+        external
+    {
+        // requires loanToken approved
+        require(supportedTokens[loanToken], "invalid");
```

Comment:
Convention for require error statements I think we should adopt is to have a prefix that is the contract's name. This makes debugging incredibly easier

Reply:
if error message fits into bytes32 its incredibly cheap. so we have to try to fit into that? do you have a format in mind for errors ?

Reply:
Since it increases gas costs, we could instead just have a table in our docs that lists what each error code is and where it occurs so it is easiest to debug.